### ISSUE TITLE
Small issue in the example of fetching a list of users

### ISSUE THREAD

**Issue by mrhdias** [ryanc-me/odoo-api-rs#1]
In the example when fetch a list of users, if the "jvec" is empty, it returns an error saying that an argument is missing.
Code to reproduce the issue:
```rust
use odoo_api::{jmap, jvec, OdooClient};
use tokio;

async fn odoo() {
    // build the client and authenticate
    let url = "https://odoo.example.com";
    let mut client = OdooClient::new_reqwest_async(url)
        .unwrap()
        .authenticate(
            "odoodbname",
            "manager@example.com",
            "mypassword",
        )
        .await
        .unwrap();

    // fetch server version info
    let version_info = client.common_version()
        .send()
        .await
        .unwrap();
    println!("Version Info: {:?}", version_info);

    // fetch a list of users
    let users = client
        .execute("res.users", "search", jvec![]) // <====== NOT WORK IF EMPTY
        // .execute("res.users", "search", jvec![[], []]) // <====== WORK WITH TWO ARGUMENTS
        .send()
        .await
        .unwrap();

    println!("Users: {:?}", users);

    // fetch the login and partner_id fields from user id=1
    let info = client
        .execute_kw(
            "res.users",
            "read",
            jvec![[1]],
            jmap! {
            "fields": ["login", "partner_id"]
            },
        )
        .send()
        .await
        .unwrap();

    println!("User Info: {:?}", info);
}

#[tokio::main]
async fn main() {
    odoo().await;
}
```
After cargo run, I get this message at the end: [String("BaseModel.search() missing 1 required positional argument: 'domain'")
```
Version Info: VersionResponse { server_version: "17.0", server_version_info: ServerVersionInfo { major: 17, minor: 0, micro: 0, release_level: "final", serial: 0, enterprise: Some("") }, server_serie: "17.0", protocol_version: 1 }
thread 'main' panicked at src/main.rs:34:10:
called `Result::unwrap()` on an `Err` value: JsonRpcError(JsonRpcError { code: 200, message: "Odoo Server Error", data: JsonRpcErrorData { name: "builtins.TypeError", debug: "Traceback (most recent call last):\n  File \"/opt/odoo17/odoo/odoo/http.py\", line 1770, in _serve_db\n    return service_model.retrying(self._serve_ir_http, self.env)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/service/model.py\", line 133, in retrying\n    result = func()\n             ^^^^^^\n  File \"/opt/odoo17/odoo/odoo/http.py\", line 1797, in _serve_ir_http\n    response = self.dispatcher.dispatch(rule.endpoint, args)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/http.py\", line 2001, in dispatch\n    result = self.request.registry['ir.http']._dispatch(endpoint)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/addons/base/models/ir_http.py\", line 222, in _dispatch\n    result = endpoint(**request.params)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/http.py\", line 725, in route_wrapper\n    result = endpoint(self, *args, **params_ok)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/addons/base/controllers/rpc.py\", line 155, in jsonrpc\n    return dispatch_rpc(service, method, args)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/http.py\", line 391, in dispatch_rpc\n    return dispatch(method, params)\n           ^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/service/model.py\", line 35, in dispatch\n    res = execute(db, uid, *params[3:])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/service/model.py\", line 65, in execute\n    res = execute_cr(cr, uid, obj, method, *args, **kw)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/service/model.py\", line 50, in execute_cr\n    result = retrying(partial(odoo.api.call_kw, recs, method, args, kw), env)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/service/model.py\", line 133, in retrying\n    result = func()\n             ^^^^^^\n  File \"/opt/odoo17/odoo/odoo/api.py\", line 464, in call_kw\n    result = _call_kw_model(method, model, args, kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/odoo17/odoo/odoo/api.py\", line 435, in _call_kw_model\n    result = method(recs, *args, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nTypeError: BaseModel.search() missing 1 required positional argument: 'domain'\n", message: "BaseModel.search() missing 1 required positional argument: 'domain'", arguments: [String("BaseModel.search() missing 1 required positional argument: 'domain'")], context: {} } })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
With two arguments everything works fine: .execute("res.users", "search", jvec![[], []])
```
Version Info: VersionResponse { server_version: "17.0", server_version_info: ServerVersionInfo { major: 17, minor: 0, micro: 0, release_level: "final", serial: 0, enterprise: Some("") }, server_serie: "17.0", protocol_version: 1 }
Users: ExecuteResponse { data: Array [Number(2), Number(6), Number(7)] }
User Info: ExecuteKwResponse { data: Array [Object {"id": Number(1), "login": String("__system__"), "partner_id": Array [Number(2), String("OdooBot")]}] }
```

**Comment by ryanc-me** [comment]
**Edit**: Snipped because I didn't read the issue properly.

@mrhdias thank you for the report - you're right, there was an error in the example! I found that, for our 16.0 instances, we actually needed to pass only 1 arg to `search`, e.g., `jvec![[]]`. Odoo methods anontated with `@api.model` should behave like this - they are model-methods, so the `self` arg is useless/not needed.

It's possible this behaviour was different in older Odoo versions. Are you able to test that theory from your side?

This new example should fetch only active users who are not OdooBot:
```rust
let users = client.execute(
    "res.users",
    "search",
    jvec![
        [["active", "=", true], ["login", "!=", "__system__"]]
    ]
).send().await?;
```
